app.js

```diff
@@ -26,10 +66,17 @@ app.use(function(req, res, next) {
 });
 
 app.get('/validate', function (req, res, next) {
-  const token = firebase.auth().createCustomToken(req.query.token);
-  res.json({ token });
+  var apiToken = req.query.token;
+  if (isValidToken(apiToken)) {
+    getApiUser(apiToken).then(function() {
+      const token = firebase.auth().createCustomToken(req.query.uid);
```

Comment:
But it probably makes more sense not to pass the user's login with the request and to use the it here. Is that what you mean?